src/data_hub/dataframe.py

```diff
@@ -0,0 +1,219 @@
+
+        date_str = self.attrs["last_updated_row"].date_updated
+        date_obj = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f").replace(
+            tzinfo=timezone.utc
+        )
+        edited_time_ago = humanize.naturaltime(now - date_obj)
+
+        header = f'<h4>{name} <a href = "{url}">🔗</a></h4>'
+        txt = f'<p style="font-size: 12px; color: #808080;">Created {created_time_ago}. Row {self.attrs["last_updated_row"].hid} was last edited {edited_time_ago}'
+        try:
+            txt += (
+                "  by "
+                + self.attrs["last_updated_row"].edited_by_user_drn.split("|")[1]
+                + ".</p>"
+            )
+        except Exception:
+            txt += ".</p>"
+
+        if self._modified_columns:
+            txt += '<p style="color: #808080; font-size: 12px">⚠️ This dataframe contains changes that have not been written back to the Deep Origin database.</p>'
+        elif self.auto_sync:
+            txt += '<p style="color: #808080; font-size: 12px">🧬 This dataframe will automatically write changes made to it back to Deep Origin.</p>'
+        df_html = super()._repr_html_()
+        return header + txt + df_html
+
+    def __repr__(self):
+        """method override to customize printing in an interactive session"""
+
+        header = f'{self.attrs["metadata"]["hid"]}\n'
+        df_representation = super().__repr__()
+        return header + df_representation
+
+    @classmethod
+    def from_deeporigin(
+        cls,
+        database_id: str,
+        *,
+        use_file_names: bool = True,
+        reference_format: IDFormat = "human-id",
+        return_type: DatabaseReturnType = "dataframe",
+        client=None,
+    ):
+        """Create a local Deep Origin DataFrame from a Deep Origin database.
+
+        Args:
+        database_id (str): The ID of the Deep Origin database.
+        use_file_names (bool, optional): Whether to use the file names in the Deep Origin database. Defaults to True.
+        reference_format (IDFormat, optional): The format of the IDs in the Deep Origin database. Defaults to "human-id".
+        return_type (DatabaseReturnType, optional): The type of return value. Defaults to "dataframe".
+
+        """
+
+        return api.get_dataframe(
+            database_id=database_id,
+            use_file_names=use_file_names,
+            reference_format=reference_format,
+            return_type=return_type,
+            client=client,
+        )
+
+    def to_deeporigin(
+        self,
+        *,
+        columns: Optional[list] = None,
+        rows: Optional[list] = None,
+    ):
+        """Write data in dataframe to Deep Origin
+
+        !!! tip "Deep Origin DataFrames automatically synchronize"
+            Typically, you do not need to manually synchronize. If the `auto_sync` attribute of the dataframe is set to `True`, the dataframe will automatically synchronize when changes are made to the dataframe.
```

Comment:
With the change to auto-sync default=False, I suggest rephrasing to 

> To automatically save changes to local DataFrames to Deep Origin databases, set the `auto_sync` attribute of the dataframe `True`.

Reply:
fixed